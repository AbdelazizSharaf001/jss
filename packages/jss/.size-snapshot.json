{
  "dist/jss.js": {
    "bundled": 77413,
    "minified": 21696,
    "gzipped": 6419
  },
  "dist/jss.min.js": {
    "bundled": 77413,
    "minified": 21696,
    "gzipped": 6419
  }
}
