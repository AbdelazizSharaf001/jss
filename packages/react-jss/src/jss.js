// @flow
import {create} from 'jss'
import preset from 'jss-preset-default'

export const jss = create(preset())
