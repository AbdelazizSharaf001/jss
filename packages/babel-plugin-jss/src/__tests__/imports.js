import {stripIndent} from 'common-tags'
import {sheets} from 'jss'
import {transform} from './testUtils'

describe('imports', () => {
  beforeEach(() => {
    sheets.reset()
  })

  test('bail out on property access from imports', () => {
    const code = stripIndent`
      import jss from 'jss';
      import config from 'config';
      createStyleSheet({
        a: {
          color: config.primary
        }
      });
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('partially bail out per property on property access from imports', () => {
    const code = stripIndent`
      import jss from 'jss';
      import config from 'config';
      createStyleSheet({
        a: {
          width: 1,
          color: config.primary
        }
      });
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('partially bail out per rule on property access from imports', () => {
    const code = stripIndent`
      import jss from 'jss';
      import config from 'config';
      createStyleSheet({
        a: {
          width: 1,
        },
        b: {
          color: config.primary
        }
      });
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('unresolvable styles ref', () => {
    const code = stripIndent`
      import jss from 'jss';
      import styles from 'styles';
      createStyleSheet(styles);
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('extend imported options ref', () => {
    const code = stripIndent`
      import jss from 'jss';
      import options from 'options';
      createStyleSheet({
        a: {
          color: 'red'
        }
      }, options);
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test.skip('resolve imports avilable modules?', () => {})
})
