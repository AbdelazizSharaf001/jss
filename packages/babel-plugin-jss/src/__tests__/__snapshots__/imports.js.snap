// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imports bail out on property access from imports 1`] = `
"[v6]
import jss from 'jss';
import config from 'config';
createStyleSheet({
  a: {
    color: config.primary
  }
});
[v7]
import jss from 'jss';
import config from 'config';
createStyleSheet({
  a: {
    color: config.primary
  }
});"
`;

exports[`imports extend imported options ref 1`] = `
"[v6]
import jss from 'jss';
import options from 'options';
createStyleSheet({
  '@raw': '.a-id {\\\\n  color: red;\\\\n}'
}, Object.assign({
  'classes': {
    'a': 'a-id'
  }
}, options));
[v7]
import jss from 'jss';
import options from 'options';
createStyleSheet({
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
}, Object.assign({
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
}, options));"
`;

exports[`imports partially bail out per property on property access from imports 1`] = `
"[v6]
import jss from 'jss';
import config from 'config';
createStyleSheet({
  '@raw': '.a-id {\\\\n  width: 1;\\\\n}',

  a: {
    color: config.primary
  }
}, {
  'classes': {
    'a': 'a-id'
  }
});
[v7]
import jss from 'jss';
import config from 'config';
createStyleSheet({
  \\"@raw\\": \\".a-id {\\\\n  width: 1;\\\\n}\\",
  a: {
    color: config.primary
  }
}, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;

exports[`imports partially bail out per rule on property access from imports 1`] = `
"[v6]
import jss from 'jss';
import config from 'config';
createStyleSheet({
  '@raw': '.a-id {\\\\n  width: 1;\\\\n}',

  b: {
    color: config.primary
  }
}, {
  'classes': {
    'a': 'a-id',
    'b': 'b-id'
  }
});
[v7]
import jss from 'jss';
import config from 'config';
createStyleSheet({
  \\"@raw\\": \\".a-id {\\\\n  width: 1;\\\\n}\\",
  b: {
    color: config.primary
  }
}, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\",
    \\"b\\": \\"b-id\\"
  }
});"
`;

exports[`imports unresolvable styles ref 1`] = `
"[v6]
import jss from 'jss';
import styles from 'styles';
createStyleSheet(styles);
[v7]
import jss from 'jss';
import styles from 'styles';
createStyleSheet(styles);"
`;
